subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -191,6 +202,82 @@ public void resolveArtifacts(ConfigurationInternal configuration, ResolverResult
         results.artifactsResolved(new DefaultResolvedConfiguration(result), result);
     }
 
+    @Override
+    public List<ArtifactRepository> getRepositories() {
+        List<ExposableRepository> exposableRepositories = CollectionUtils.collect(repositories, Transformers.cast(ExposableRepository.class));
```

Comment:
This is unsafe. If one of the repositories is not exposable, this will blow up at runtime.

Reply:
:x: +1. And this method is going to be called multiple times, and is pretty expensive. You should cache the result in a field. You don't have to bother with thread-safety here, it's no big deal if 2 threads recompute the same thing concurrently.

Reply:
done and cached